compute_endpoint/globus_compute_endpoint/endpoint/endpoint.py

```diff
@@ -390,12 +391,21 @@ def start_endpoint(
                 )
 
             except GlobusAPIError as e:
-                if e.http_status in (409, 410, 423):
-                    # CONFLICT, GONE or LOCKED
-                    blocked_msg = f"Endpoint registration blocked.  [{e.text}]"
+                blocked_msg = f"Endpoint registration blocked.  [{e.text}]"
+                log.warning(blocked_msg)
+                if e.http_status in (
+                    HTTPStatus.CONFLICT,
+                    HTTPStatus.LOCKED,
+                    HTTPStatus.NOT_FOUND,
+                ):
                     print(blocked_msg)
-                    log.warning(blocked_msg)
                     exit(os.EX_UNAVAILABLE)
+                elif e.http_status in (
+                    HTTPStatus.BAD_REQUEST,
+                    HTTPStatus.UNPROCESSABLE_ENTITY,
+                ):
+                    print(blocked_msg)
+                    exit(os.EX_DATAERR)
```

Comment:
I would include the `print(blocked_msg)` for all paths.  Regardless of whether we handle the specific failure case, the key _outcome_ is that upstream did not honor the request.  The traceback is mostly a measure for us to help debug the situation if it makes it back to us (e.g., via #help).